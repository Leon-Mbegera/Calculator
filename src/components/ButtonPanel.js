import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import './calc.css';

function clickButton(label, handleClick) {
  return <Button show={label} onClick={() => handleClick(label)} />;
}

function ButtonPanel({ clickHandler }) {
  const handleClick = (buttonName) => {
    clickHandler(buttonName);
  };

  return (
    <>
      <div className="button-panel">
        <div className="panel-row">
          {clickButton('AC', handleClick)}
          {clickButton('+/-', handleClick)}
          {clickButton('%', handleClick)}
          {clickButton('รท', handleClick)}
        </div>
        <div>
          {clickButton('7', handleClick)}
          {clickButton('8', handleClick)}
          {clickButton('9', handleClick)}
          {clickButton('x', handleClick)}
        </div>
        <div>
          {clickButton('4', handleClick)}
          {clickButton('5', handleClick)}
          {clickButton('6', handleClick)}
          {clickButton('-', handleClick)}
        </div>
        <div>
          {clickButton('1', handleClick)}
          {clickButton('2', handleClick)}
          {clickButton('3', handleClick)}
          {clickButton('+', handleClick)}
        </div>
        <div>
          {clickButton('0', handleClick)}
          {clickButton('.', handleClick)}
          {clickButton('=', handleClick)}
        </div>
      </div>
    </>
  );
}

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
