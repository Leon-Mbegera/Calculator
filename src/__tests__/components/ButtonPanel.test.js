import renderer from 'react-test-renderer';
import ButtonPanel from '../../components/ButtonPanel';

test('renders Display and ButtonPanel components', () => {
  const snap = renderer.create(<ButtonPanel />).toJSON();
  expect(snap).toMatchSnapshot();
});
