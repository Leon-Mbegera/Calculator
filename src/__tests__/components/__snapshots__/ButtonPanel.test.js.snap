// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Display and ButtonPanel components 1`] = `
<div
  className="button-panel"
>
  <div
    className="panel-row"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "lavender",
          "width": "25%",
        }
      }
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="panel-row"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "lavender",
          "width": "25%",
        }
      }
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="panel-row"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "lavender",
          "width": "25%",
        }
      }
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="panel-row"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "lavender",
          "width": "25%",
        }
      }
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="panel-row"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "50%",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "whitesmoke",
          "width": "25%",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "background": "lavender",
          "width": "25%",
        }
      }
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
